### PR TITLE
Kube Rebase 1.30.2

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
1. Updates golang to 1.22 everywhere
2. Updates K8s to 1.30.2 everywhere

updating to 1.30.2 as that is the latest kindtest/node image available 

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by martinkennelly** [comment]
@JacobTanenbaum let me know when its ready for review - dm me.

File: test/e2e/go.mod [hunk 1]
```diff
@@ -154,7 +154,7 @@ require (
 	github.com/docker/docker v24.0.9+incompatible
 	github.com/google/goexpect v0.0.0-20210430020637-ab937bf7fd6f
 	github.com/onsi/ginkgo v1.16.5
-	github.com/openshift-kni/k8sreporter v1.0.4
```
Comment (martinkennelly) [inline test/e2e/go.mod:156]: why did you need to upgrade to version that no longer uses glog?

File: go-controller/go.mod [hunk 2]
```diff
@@ -60,7 +60,7 @@ require (
 	k8s.io/kubernetes v1.30.0
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
 	kubevirt.io/api v1.0.0-alpha.0
-	sigs.k8s.io/controller-runtime v0.17.0
+	sigs.k8s.io/controller-runtime v0.18.0
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/go.mod:63]: It is mentioned in the release description for the tag -  https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.18.0

File: test/e2e/go.mod (hunk 1 above)
Comment (JacobTanenbaum) [reply test/e2e/go.mod:156]: this will get a little into the weeds of golang but here goes. A flag can only be defined once globally, we have two packages that define the "v" flag 

1. github.com/golang/glog
2. k8s.io/component-base

they have both existed fine until now because nothing imported "k8s.io/component-base/logs/testinit" which is what sets the "v" flag for the component-base package. Until kube 1.30 which is the first release that imports "k8s.io/component-base/logs/testinit" in "k8s.io/kubernetes/test/e2e/framework/ginkgologger.go" so now both packages try to define the same flag. So I have to find a way to remove glog if we want to continue to use anything in "k8s.io/kubernetes/test/e2e/framework"

File: test/e2e/go.mod [hunk 3]
```diff
@@ -1,22 +1,24 @@
 module github.com/ovn-org/ovn-kubernetes/test/e2e
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.2
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply test/e2e/go.mod]: because I didn't realize that I could change it. fixed

**Comment by martinkennelly** [comment]
/lgtm

you you inspect the CI test failures and see if theyre flakes and if so link the issue here, if no issue, create a new one.

**Comment by JacobTanenbaum** [comment]
CI failures are flakes 
https://github.com/ovn-org/ovn-kubernetes/issues/4432
https://github.com/ovn-org/ovn-kubernetes/issues/4144

**Comment by JacobTanenbaum** [comment]
@tssurya, @trozet, @jcaamano could an approver PTAL when they get the chance?

File: go-controller/go.mod [hunk 4]
```diff
@@ -50,15 +50,15 @@ require (
 	gopkg.in/fsnotify/fsnotify.v1 v1.4.7
 	gopkg.in/gcfg.v1 v1.2.3
 	gopkg.in/natefinch/lumberjack.v2 v2.2.1
-	k8s.io/api v0.29.2
-	k8s.io/apimachinery v0.29.2
-	k8s.io/client-go v0.29.2
+	k8s.io/api v0.30.0
+	k8s.io/apimachinery v0.30.0
+	k8s.io/client-go v0.30.0
```
Comment (jcaamano) [inline go-controller/go.mod:55]: Do you need to bump `k8s.io/component-helpers v0.29.2` below as well?

File: go-controller/cmd/ovnkube/ovnkube.go [hunk 5]
```diff
@@ -620,11 +620,13 @@ func (m leaderMetrics) Off(string) {
 	}
 }
 
+func (m leaderMetrics) SlowpathExercised(string) {}
+
```
Comment (jcaamano) [inline go-controller/cmd/ovnkube/ovnkube.go:624]: @martinkennelly this could potentially take a metric, seems like some kind of slow path leader acquire.

File: test/e2e/go.mod [hunk 6]
```diff
@@ -146,20 +145,19 @@ require (
...
+	k8s.io/component-helpers v0.30.0 // indirect
+	k8s.io/controller-manager v0.30.0 // indirect
+	k8s.io/klog/v2 v2.120.1 // indirect
+	k8s.io/kms v0.30.0 // indirect
+	k8s.io/kube-openapi v0.0.0-20240228011516-70dd3763d340 // indirect
+	k8s.io/kubelet v0.30.0 // indirect
 	kubevirt.io/containerized-data-importer-api v1.57.0-alpha1 // indirect
 	kubevirt.io/controller-lifecycle-operator-sdk/api v0.0.0-20220329064328-f3cc58c6ed90 // indirect
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.28.0 // indirect
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.29.0 // indirect
```
Comment (jcaamano) [inline test/e2e/go.mod:160]: This could potentially be bumped explicitly

**Review by tssurya — COMMENTED** [review]
ah realised I had some stale comments pending...

File: contrib/kind.sh [hunk 7]
```diff
@@ -492,7 +492,7 @@ set_default_params() {
   fi
   RUN_IN_CONTAINER=${RUN_IN_CONTAINER:-false}
   KIND_IMAGE=${KIND_IMAGE:-kindest/node}
-  K8S_VERSION=${K8S_VERSION:-v1.29.2}
+  K8S_VERSION=${K8S_VERSION:-v1.30.0}
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply contrib/kind.sh]: I don't know, how would I figure that out?

File: go-controller/go.mod
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/go.mod:1]: OCP projects seem to move to whatever the highest version is when they bumped, I changed this to bump to 1.30.3

File: go-controller/go.mod [hunk 8]
```diff
@@ -28,7 +28,7 @@ require (
 	github.com/miekg/dns v1.1.31
 	github.com/mitchellh/copystructure v1.2.0
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.30.0
+	github.com/onsi/gomega v1.31.0
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/go.mod]: yes, kubernetes 1.30.x depends on gomega v1.31.0

File: go-controller/go.mod [hunk 9]
```diff
@@ -1,6 +1,6 @@
 module github.com/ovn-org/ovn-kubernetes/go-controller
 
-go 1.21
+go 1.22.0
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/go.mod:3]: Is it? I see it in many of the go.mods [kubernetes](https://github.com/kubernetes/kubernetes/blob/e1aa8197eddd277fb82be0b8bbc2ba2b4ca67af7/go.mod#L9), [client-go](https://github.com/kubernetes/client-go/blob/5130bd9e786252d41e7d12e32b76355620dd26a7/go.mod#L5), [openshift/intstaller](https://github.com/openshift/installer/blob/574801453f752049722f02e3de0800ef1cba37d7/go.mod#L3),  and [origin](https://github.com/openshift/origin/blob/fe2c8559d0707594639538987b1ac33ade5388c1/go.mod#L3)

File: test/e2e/go.mod (hunk 6 above)
Comment (JacobTanenbaum) [reply test/e2e/go.mod]: I bumped this explicitly

File: go-controller/go.mod (hunk 4 above)
Comment (JacobTanenbaum) [reply go-controller/go.mod]: turns out we don't need it but I bumped it explicitly to align

File: go-controller/go.mod (hunk 9 above)
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/go.mod:3]: Is it? I see it in many of the go.mods [kubernetes](https://github.com/kubernetes/kubernetes/blob/e1aa8197eddd277fb82be0b8bbc2ba2b4ca67af7/go.mod#L9), [client-go](https://github.com/kubernetes/client-go/blob/5130bd9e786252d41e7d12e32b76355620dd26a7/go.mod#L5), [openshift/intstaller](https://github.com/openshift/installer/blob/574801453f752049722f02e3de0800ef1cba37d7/go.mod#L3),  and [origin](https://github.com/openshift/origin/blob/fe2c8559d0707594639538987b1ac33ade5388c1/go.mod#L3)

File: go-controller/go.mod
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/go.mod:1]: I changed it again to 1.30.2 because that is the latest kindtest/node image that is available

**Comment by JacobTanenbaum** [comment]
CI failure is this flake again: https://github.com/ovn-org/ovn-kubernetes/issues/4144

File: contrib/kind.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply contrib/kind.sh]: we talked about this offline there is no work to be done

**Comment by JacobTanenbaum** [comment]
CI issue is this flake: https://github.com/ovn-org/ovn-kubernetes/issues/4432

**Review by jcaamano — APPROVED** [review]
